core/src/main/java/io/questdb/jit/CompiledFilterIRSerializer.java

```diff
@@ -980,6 +987,9 @@ public int constantTypeCode() {
         }
 
         public boolean hasMixedSizes() {
+            if (sizes[STR_INDEX] != 0) {
+                return true;
```

Comment:
Although scalar mode is always suboptimal, at least we know this won't slow down previous queries, because a query involving a varlen column has so far been inelligible for JIT compilation.